src/cmd_line/subparsers/setoptions.ts

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+import * as node from "../commands/setoptions";
+import {Scanner} from '../scanner';
+
+export function parseOption(args: string) : node.IOptionArgs {
+  let scanner = new Scanner(args);
+  scanner.skipWhiteSpace();
+
+  if (scanner.isAtEof) {
+    return {};
+  }
+
+  let option = scanner.nextWord();
+
+  if (option.startsWith("no")) {
+    // :se[t] no{option}  Toggle option: Reset, switch it off.
+    return {
+      name: option.substring(2, option.length),
+      operator: node.SetOptionOperator.Reset
+    };
+  }
+
+  if (option.includes('=')) {

```

Comment:
"foo=bar" should not parse as a single word, but rather 3 separate words: ["foo", "=", "bar"]. Same with ":".

I realize Vim has some slightly bizarre rules about whitespace, but I think it's okay to be more permissive than they are if it is also easier to code it that way.